check/mypy

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+################################################################################
+# Runs mypy on the repository using a preconfigured mypy.ini file.
+#
+# Usage:
+#     check/mypy [--flags]
+################################################################################
+
+# Get the working directory to the repo root.
+cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
Why isn't this equivalent to
```
cd "$( dirname "${BASH_SOURCE[0]}" )"
```
?

Reply:
Because of the `&& pwd` at the end.

Reply:
That just gets your present working directory, which you're already in, right? Why do you need to get it, and then cd into it if you're already there?

Reply:
Oh, that. When you invoke something like `$(expression)` it runs in a new bash instance, so the `cd` applies in the new instance but not in the instance that then runs the outer `cd`.

https://stackoverflow.com/a/246128/52239

Reply:
Sorry, I still don't see why you can't let the `cd` in
```
cd "$( dirname "${BASH_SOURCE[0]}" )"
```
be the outer `cd` and be done with it.

Reply:
You removed the `&& pwd` part

Reply:
Okay clearly I'm not a bash expert and sorry to be annoying about this unimportant detail, but I do want to understand this. My understanding is that
```
cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
means "change directory to the directory that contains this file."
On the other hand, it appears to me that
```
cd "$( dirname "${BASH_SOURCE[0]}" )"
```
also means "change directory to the directory that contains this file." So I believe (probably incorrectly) that both of these statements have the same effect in all situations.

Reply:
I think you're right. Changed it.

I took this from SO where the context was "I want the path as a string". But clearly if you're just feeding it back into cd then using pwd to clean it up, cd already had to understand it. So we don't need the cleaned up path, just the original one.